index.js

```diff
@@ -4,12 +4,29 @@
 module.exports = {
     name: 'ember-cli-hot-loader',
     serverMiddleware: function (config){
-        var lsReloader = require('./lib/hot-reloader')(config.options);
-        lsReloader.run();
+        if (config.options.environment === 'development') {
+            var lsReloader = require('./lib/hot-reloader')(config.options);
+            lsReloader.run();
+        }
     },
     included: function (app) {
         this._super.included(app);
-        // TODO: consider removing this since it adds an unnecessary runtime dependency to all apps
-        app.import(app.bowerDirectory + '/ember/ember-template-compiler.js');
+
+        // If not in dev, bail
+        if (app.env !== 'development') {
+            return;
+        }
+
+        var bowerPath = app.bowerDirectory + '/ember/ember-template-compiler.js';
+        var npmPath = app.project.nodeModulesPath + '/ember-source/dist/ember-template-compiler.js';
+
+        // Require template compiler as in CLI this is only used in build, we need it at runtime
+        if (fs.existsSync(bowerPath)) {
```

Comment:
I'm not familiar with this, le tme know if you need this changing, or alternatively cut a separate PR once this has merged.